pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/stats/AbstractColumnStatisticsCollector.java

```diff
@@ -120,4 +141,48 @@ Object addressNull(Object entry, DataType e) {
     }
     return entry;
   }
+
+  /**
+   * Returns the {@link PartitionFunction} for the column.
+   * @return Partition function for the column.
+   */
+  public PartitionFunction getPartitionFunction() {
+    return partitionFunction;
+  }
+
+  /**
+   * Returns the partition ranges within which the column values exist.
+   *
+   * @return List of ranges for the column values.
+   */
+  public List<IntRange> getPartitionRanges() {
+    return partitionValues;
+  }
+
+  /**
+   * Checks if the given value lies within one of the partition ranges. If the value is not within
+   * any of the partition ranges, partitioning is dropped. The effect of that is:
+   * <ul>
+   *   <li> Subsequent values will not be checked against partition ranges. </li>
+   *   <li> Partition information will not be written out to the metadata for this column. </li>
+   * </ul>
+   *
+   * @param value Column value to check.
+   */
+  protected void checkPartition(Object value) {
+    if (partitionFunction != null) {
+      int partition = partitionFunction.getPartition(value);
+
+      for (IntRange partitionValue : partitionValues) {
+        if (partition >= partitionValue.getMinimumInteger() && partition <= partitionValue.getMaximumInteger()) {
+          return;
```

Comment:
Yes, the log is a few lines below. This return is the happy path when value is in range, ie no action needed. When value is not in range, we clear the partition function from stats collector, so it does not get written to metadata.